### PR TITLE
add redux selectors

### LINKED ISSUE #96: Add redux selectors
It would be great to have memoized selectors to compute derived data from the redux store https://redux.js.org/recipes/computingderiveddata. Not only for computing derived data, but also to stop doing toJS calls on the mapState function, that are triggering re-renders because of the nature of the toJS method, and the shallow compare that React-Redux does

### PR DESCRIPTION
**Pull request by pedrozs**
# add redux selectors
## Description
This PR resolves #96.
It also adds the username to the homepage message, to give an example of a selector.

### CODE REVIEW

File: src/containers/HomePage.js [hunk 1]
```diff
@@ -1,13 +1,18 @@
...
+    <p><FormattedMessage id="home.welcome" />{` ${username}`}</p>
     <LogoutButton />
   </div>
 );
 
-export default HomePage;
+const mapState = state => ({
+  username: getUser(state).username
+});
+export default connect(mapState)(HomePage);
```
Comment (bernabe9) [inline src/containers/HomePage.js]: add an empty line above

File: src/containers/HomePage.js [hunk 2]
```diff
@@ -1,13 +1,18 @@
...
+import { connect } from 'react-redux';
 
+import { getUser } from 'selectors/sessionSelectors';
 import LogoutButton from 'components/user/LogoutButton';
 
-const HomePage = () => (
+const HomePage = username => (
   <div>
-    <p><FormattedMessage id="home.welcome" /></p>
+    <p><FormattedMessage id="home.welcome" />{` ${username}`}</p>
```
Comment (bernabe9) [inline src/containers/HomePage.js]: Why are you adding a space before the username? 🤔

Comment (pedrozs) [reply src/containers/HomePage.js]: To add a space between the welcome message and the username. I've added it as a value now anyway.

File: src/locales/es.js [hunk 3]
```diff
@@ -19,7 +19,7 @@ export default {
 
   'logout.button': 'LOGOUT',
 
-  'home.welcome': 'Welcome to React Redux Base',
+  'home.welcome': 'Welcome to React Redux Base {username}',
```
Comment (missx) [inline src/locales/es.js]: Shouldn't this be in Spanish?

File: src/containers/HomePage.js [hunk 4]
```diff
@@ -1,13 +1,19 @@
 import React from 'react';
 import { FormattedMessage } from 'react-intl';
+import { connect } from 'react-redux';
 
+import { getUser } from 'selectors/sessionSelectors';
 import LogoutButton from 'components/user/LogoutButton';
 
-const HomePage = () => (
+const HomePage = username => (
```
Comment (fsarachu) [inline src/containers/HomePage.js]: Aren't you missing the destructuring `({username})`? 🤔

File: src/locales/es.js (hunk 3 above)
Comment (pedrozs) [reply src/locales/es.js]: It is now.

**Pull request merged**